### DISCUSSION TITLE
Sync between variables / style and token set / tokens

### DISCUSSION THREAD

**Discussion by ndupasquier** [primefaces/community#3218] [Figma UI Kit] [answered]
Hello 
I'm using the v 3.0.1 UI KIT, problem is that when i export styles and variables from token studio to figma to update my variables, it creats new collection of variables instead of updating existing one. Here are my figma variables before, and after you can see that all collections and variable has been recreated instead of update existing one
**The expected behavior is that the tokens inside each token set update the existing variable inside the "Aura" collection of figma.**

Before i export the variables
<img width="1083" alt="Capture d’écran 2024-12-14 à 09 29 53" src="https://github.com/user-attachments/assets/fb24ce0d-5a85-47d4-8ff5-a372940e871c" />

After i export the variables, new collections appear instead of updating existing variables
<img width="598" alt="Capture d’écran 2024-12-14 à 09 31 02" src="https://github.com/user-attachments/assets/8c6785e7-0a57-4fc0-adf9-1c7ae9a89807" />

Can you explain how to update existing variables from token studio and vice versa ? update token studio from figma variables ?

**Comment by onursenture** [comment, marked as answer]
Hello, Some other users reported the same issue lately, so I recorded a screen capture as an experiment. I recommend following the steps shown in the video. You need to use Tokens Studio as the primary token control tool. There isn't a method to update from Figma Variables to Tokens Studio.

https://github.com/user-attachments/assets/5fc9b376-5d89-4eee-98ea-981214ebe6d6

PS: You need to be sure the Themes tab is selected while exporting, even if you're using the free version of Tokens Studio.

**Reply by gonzalodisa** [reply]
Hello,

I have a problem when i export the tokens from the plugin to the Local Variables. 

I only change the primary color palette in the page "semantic" in Token studio, and doing this I can see all the changes well in the plugin. But when i follow the steps of the video that you attached, i loose all the color variables in the figma local variables. Also the color tokens i dont touch.

Why? I have to change manualy the tokens in the figma local variables? Why this happens?

Original file:
![image](https://github.com/user-attachments/assets/a256d168-8881-4fe8-a575-4c1163b1bfcb)

Edited file:
![image](https://github.com/user-attachments/assets/5cb76f8e-6bdb-4dc1-898d-5e1f64e2b0e3)


Regards

**Reply by onursenture** [reply]
> Hello,
> 
> I have a problem when i export the tokens from the plugin to the Local Variables.
> 
> I only change the primary color palette in the page "semantic" in Token studio, and doing this I can see all the changes well in the plugin. But when i follow the steps of the video that you attached, i loose all the color variables in the figma local variables. Also the color tokens i dont touch.
> 
> Why? I have to change manualy the tokens in the figma local variables? Why this happens?
> 
> Original file: ![image](https://private-user-images.githubusercontent.com/197061604/408139463-a256d168-8881-4fe8-a575-4c1163b1bfcb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgyNDU4OTYsIm5iZiI6MTczODI0NTU5NiwicGF0aCI6Ii8xOTcwNjE2MDQvNDA4MTM5NDYzLWEyNTZkMTY4LTg4ODEtNGZlOC1hNTc1LTRjMTE2M2IxYmZjYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEzMFQxMzU5NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kNDdlM2U0MzQ5M2JmYzEzNTQwZjBjZDliYjFmZjc3YmEwMTYwNjc3MjkyNjk3NjM2MWViNzI5YWY1NDc2N2ViJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.VvlmGa5iX1pG3Lf6ZVeNGdC2Pu3FJJsrVJmgs5mNiCI)
> 
> Edited file: ![image](https://private-user-images.githubusercontent.com/197061604/408139405-5cb76f8e-6bdb-4dc1-898d-5e1f64e2b0e3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgyNDU4OTYsIm5iZiI6MTczODI0NTU5NiwicGF0aCI6Ii8xOTcwNjE2MDQvNDA4MTM5NDA1LTVjYjc2ZjhlLTZiZGItNGRjMS04OThkLTVlMWY2NGUyYjBlMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEzMFQxMzU5NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNmI0Yjk0OTljODNhNzZlNzkyNGFkYjQ2NmFkZTA2YjllZTc5MzJkNWM2NDFlZTJjODk4NDk4MTcwMjE1ZmIxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.RFazp14wy7fxxGxStkvPDXeYJqNplzY71XF_kUhFSgg)
> 
> Regards

Yes, I experienced the same issue when I followed the same steps. I didn't encounter this problem before; perhaps something has changed on Tokens Studio's end. 

On my second attempt, I selected all the checkboxes under "Create and Update," and all the connections were fixed. This might be an intermittent bug that resolves when an action is repeated, or perhaps the export algorithm has changed from previous versions. 

I recommend trying again, making sure to select all the checkboxes under "Create and Update."

<img width="379" alt="image" src="https://github.com/user-attachments/assets/59ec8182-2a61-4a52-818d-ecee9bb41c55" />

**Reply by onursenture** [reply]
<img width="722" alt="image" src="https://github.com/user-attachments/assets/a48bcfaf-0715-44a8-915e-ddeea9f981af" />

I found the related documentation here, which might help: [Tokens Studio Export Options](https://docs.tokens.studio/variables-and-styles/export-to-figma-options-menu#3-create-styles-with-variable-references).

**Reply by gonzalodisa** [reply]
It workss!! thank you so much

**Reply by rupserinus** [reply]
Thanks! Had the same issue that Token Studio created a lot of sets instead of updating the token in local variables

**Reply by aduda9** [reply]
Just in case anyone is missing this like I did - this step seems important (make sure you're currently on this tab when you hit export)

> PS: You need to be sure the Themes tab is selected while exporting, even if you're using the free version of Tokens Studio.
